// @remotion/tailwind/.rollup.cache/Users/jonathanburger/remotion/packages/tailwind/dist/esm/enable.js
/**
 * @description A function that modifies the default Webpack configuration to make the necessary changes to support Tailwind.
 * @see [Documentation](https://www.remotion.dev/docs/tailwind/enable-tailwind)
 */
export const enableTailwind = (currentConfiguration) => {
    var _a;
    return {
        ...currentConfiguration,
        module: {
            ...currentConfiguration.module,
            rules: [
                ...(((_a = currentConfiguration.module) === null || _a === void 0 ? void 0 : _a.rules)
                    ? currentConfiguration.module.rules
                    : []).filter((rule) => { var _a; return rule !== '...' && !((_a = rule.test) === null || _a === void 0 ? void 0 : _a.toString().includes('.css')); }),
                {
                    test: /\.css$/i,
                    use: [
                        require.resolve('style-loader'),
                        require.resolve('css-loader'),
                        {
                            loader: require.resolve('postcss-loader'),
                            options: {
                                postcssOptions: {
                                    plugins: [
                                        require.resolve('postcss-preset-env'),
                                        require.resolve('tailwindcss'),
                                        require.resolve('autoprefixer'),
                                    ],
                                },
                            },
                        },
                    ],
                },
            ],
        },
    };
};
